debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -64,26 +75,61 @@ public static String getClassName(J9IndexableObjectPointer objPointer) throws Co
 		return J9ArrayClassHelper.getName(clazz(objPointer));
 	}
 
-	public static U32 size(J9IndexableObjectPointer objPointer) throws CorruptDataException 
+	public static U32 rawSize(J9IndexableObjectPointer objPointer) throws CorruptDataException 
 	{
-		UDATA size = J9IndexableObjectContiguousPointer.cast(objPointer).size();
-
-		if (size.eq(0)) {
-			size = J9IndexableObjectDiscontiguousPointer.cast(objPointer).size();
+		if (mixedModeSupport) {
+			if (J9JavaVMHelper.compressObjectReferences) {
+				return J9IndexableObjectContiguousCompressedPointer.cast(objPointer).size();
+			}
+			return J9IndexableObjectContiguousFullPointer.cast(objPointer).size();
 		}
+		return J9IndexableObjectContiguousPointer.cast(objPointer).size();
+	}
 
-		if (size.anyBitsIn(0x80000000)) {
```

Comment:
What's the justification for removing this check?

Reply:
rawSize returns the untreated value of the contiguous size field. The check is done in the array processing code in the GC - I could add it back in to the size method below.

Reply:
I think it's worth adding back.

Reply:
Done.